// @aztec/bb.js/dest/node-cjs/barretenberg/verifier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BarretenbergVerifier = void 0;
const index_js_1 = require("./index.js");
const raw_buffer_js_1 = require("../types/raw_buffer.js");
const index_js_2 = require("../proof/index.js");
// TODO: once UP is removed we can just roll this into the bas `Barretenberg` class.
class BarretenbergVerifier {
    constructor(options = { threads: 1 }) {
        this.options = options;
    }
    /** @ignore */
    async instantiate() {
        if (!this.api) {
            const api = await index_js_1.Barretenberg.new(this.options);
            await api.initSRSForCircuitSize(0);
            this.api = api;
        }
    }
    /** @description Verifies a proof */
    async verifyUltraHonkProof(proofData, verificationKey) {
        await this.instantiate();
        const proof = (0, index_js_2.reconstructHonkProof)((0, index_js_2.flattenFieldsAsArray)(proofData.publicInputs), proofData.proof);
        return await this.api.acirVerifyUltraHonk(proof, new raw_buffer_js_1.RawBuffer(verificationKey));
    }
    async destroy() {
        if (!this.api) {
            return;
        }
        await this.api.destroy();
    }
}
exports.BarretenbergVerifier = BarretenbergVerifier;
